src/LP/LP.jl

```diff
@@ -30,8 +30,52 @@ end
 #
 # ==============================================================================
 
+
+const MAX_LENGTH = 255
+# 16 for lp_solve 5.0: http://lpsolve.sourceforge.net/5.0/CPLEX-format.htm
+# 255 for CPLEX 12.5: https://www.ibm.com/support/knowledgecenter/SS9UKU_12.5.0/com.ibm.cplex.zos.help/FileFormats/topics/LP.html
+const START_REG = r"^([\.0-9eE])"
+const NAME_REG = r"([^a-zA-Z0-9\!\"\#\$\%\&\(\)\/\,\.\;\?\@\_\`\'\{\}\|\~])"
+
+let variable_name_cache = Dict{String, String}()
+    global sanitized_name
+    function sanitized_name(name::String)
+        if name in keys(variable_name_cache)
+            return variable_name_cache[name]
+        end
+        varname = name
+
+        m = match(START_REG, name)
+        if m !== nothing
+            plural = length(m.match) > 1
+            @warn("Name $(name) cannot start with a period, a number, e, or E. " *
+                  "Prepending an underscore to name.")
+            return sanitized_name("_" * name)
+        end
+
+        m = match(NAME_REG, name)
+        if m !== nothing
+            plural = length(m.match) > 1
+            @warn("Name $(name) contains $(ifelse(plural, "", "an "))" *
+                  "illegal character$(ifelse(plural, "s", "")): " *
+                  "\"$(m.match)\". Removing the offending " *
+                  "character$(ifelse(plural, "s", "")) from name.")
+            return sanitized_name(replace(name, NAME_REG => s"_"))
+        end
+
+        # Truncate at the end to fit as many characters as possible.
+        if length(name) > MAX_LENGTH
+            @warn("Name $(name) too long (length: $(length(name))). Truncating.")
+            return sanitized_name(String(name[1:MAX_LENGTH]))
+        end
+
+        variable_name_cache[varname] = name
+        return name
+    end
+end
```

Comment:
I was more thinking along the lines of 
```Julia
sanitized_names = Dict{MOI.VariableIndex, String}()
for v in MOI.get(model, MOI.ListOfVariableIndices())
    sanitized_names[v] = sanitize_name(MOI.get(model, MOI.VariableName(), v))
end
```
and then passing `sanitized_names` to `write_function` so that we don't have to do the look-up (and sanitization) for each variable.

Reply:
I don't exactly know what this changed, but now all tests are passing!